src/web/lib/contributions.tsx

```diff
@@ -0,0 +1,92 @@
+export const shouldShowSupportMessaging = (): boolean => {
+    const hideSupportMessaging =
+        getCookie(HIDE_SUPPORT_MESSAGING_COOKIE) === 'true';
+
+    return !hideSupportMessaging;
+};
+
+// Determine if user is a recurring contributor by checking if they are signed in
+// AND have at least one of the relevant cookies.
+// We need to look at both User Attributes and Frontend Support cookies
+// as the former might not reflect the latest contributor status, since it's set upon signing in.
+// Frontend Support cookies are set when a contribution is made.
+export const isRecurringContributor = (isSignedIn: boolean): boolean => {
+    // Attributes cookie - we want this to have a specific value
+    const isRecurringContributorFromAttrs =
+        getCookie(RECURRING_CONTRIBUTOR_COOKIE) === 'true';
+
+    // Support cookies - we only care whether these exist
+    const hasMonthlyContributionCookie =
+        getCookie(SUPPORT_RECURRING_CONTRIBUTOR_MONTHLY_COOKIE) !== null;
+    const hasAnnualContributionCookie =
+        getCookie(SUPPORT_RECURRING_CONTRIBUTOR_ANNUAL_COOKIE) !== null;
+
+    return (
+        isSignedIn &&
+        (isRecurringContributorFromAttrs ||
+            hasMonthlyContributionCookie ||
+            hasAnnualContributionCookie)
+    );
+};
+
+// looks at attribute and support cookies
+// ONE_OFF_CONTRIBUTION_DATE_COOKIE (attributes cookie, when loggin in)
+// SUPPORT_ONE_OFF_CONTRIBUTION_COOKIE (support cookie, when making one-off contribution)
+// Get the date of the latest one-off contribution by looking at the two relevant cookies
+// and returning a Unix epoch string of the latest date found.
+export const getLastOneOffContributionDate = (): string => {
+    // Attributes cookie - expects YYYY-MM-DD
+    const contributionDateFromAttributes = getCookie(
+        ONE_OFF_CONTRIBUTION_DATE_COOKIE,
+    );
+
+    // Support cookies - expects Unix epoch
+    const contributionDateFromSupport = getCookie(
+        SUPPORT_ONE_OFF_CONTRIBUTION_COOKIE,
+    );
+
+    try {
+        // Parse dates into common format so they can be compared
+        const parsedDateFromAttributes = contributionDateFromAttributes
+            ? Date.parse(contributionDateFromAttributes)
+            : 0;
+        const parsedDateFromSupport = contributionDateFromSupport
+            ? parseInt(contributionDateFromSupport, 10)
+            : 0;
+
+        // Determine the latest of two dates
+        const latestOnOffContributionDate = Math.max(
+            parsedDateFromAttributes,
+            parsedDateFromSupport,
+        );
+
+        // Return stringified latest contribution date
+        return latestOnOffContributionDate
+            ? String(latestOnOffContributionDate)
+            : '';
+    } catch (e) {
+        // eslint-disable-next-line no-console
+        console.log('Could not parse latestOneOffContribution date');
```

Comment:
Do we want to write to console on error? Maybe if there's benefit in tracking these we could post to Sentry?

Reply:
The try/catch is gone now, not sure there's any place where it makes sense to send an error to Sentry.